lib/id3c/cli/command/etl/redcap_det.py

```diff
@@ -89,9 +90,16 @@ def command_for_project(name: str,
     }
 
     def decorator(routine: Callable[..., Optional[dict]]) -> click.Command:
+        @click.option("--limit",
+            metavar = "<number>",
+            help    = "Maximum number of received DETs to process in this invocation. "
+                      "If more unprocessed DETs exist, they will be processed on the next invocation of this ETL. "
+                      "Not limited by default.",
+            type    = int)
+
         @click.option("--fetch-batch-size",
```

Comment:
Could we name the two options something more specific? Something like:
--limit > --det_batch_size
--fetch-batch-size > --redcap_batch_size

Reply:
I'm not sure I follow, @jccraft-uw. 
Both of them relate to REDCap DETs, so the distinction between the suggested "det" and "redcap" prefixes doesn't register with me.

IIUC, we fetch DETs from REDCap, in given batch of size N, up to a limit M.

Reply:
They do both relate to REDCap.
limit : How many records should the job pull from the receiving.redcap_det table?
fetch_batch_size: When querying REDCap for records how many are queried in one batch?

It would make it easier to know what each option controls if we give them a more specific name.

Reply:
@jccraft-uw Fair point, and thank you for considering usability concerns like this! I'm open to using different names here, although I don't much like `--det-batch-size` and `--redcap-batch-size` as alternatives.¹ Let's find other options (har har) that will work for everyone.

Part of my thinking that lead me to `--limit` is that I anticipate we'll eventually want the same option across all `id3c etl` commands, not just those under `id3c etl redcap-det`. This was actually what I went into the codebase intending to do, but then realized that we still hadn't yet refactored our ETLs' SQL select pattern into a common decorator yet. Bummer. The point being though, is that `--limit` is the hard limit of `receiving.*` rows to process—that's only `receiving.redcap_det` for now—and the concept of a "limit" will be understood by anyone familiar with SQL (which is good, since that's exactly what its coupled to here).

When writing the code, I considered the following instead of `--limit`:

- `--batch-size`, but "batch" to me implies that something will be done repeatedly in chunks of that size. This is true for `--fetch-batch-size`, but it isn't true for the functionality I'm adding here.

- `--size` and `--count`, but these felt too vague.

- `-n`, which has CLI tradition on its side for a limiting option, but this felt too terse.

Thinking through this again now, maybe `--receiving-limit` would be more clear?

I agree that the existing `--fetch-batch-size` option isn't as clearly named as I'd like, but I struggled at the time to find alternatives that felt better. I liked the "batch size" part because of the connotations it has for me, as above. I considered `--redcap-batch-size`, but thought that was maybe confusing for some other behaviour since "redcap" could mean the DETs or the records. "Fetch" felt better for that reason, as it implies accessing a remote/external resource for me.

Maybe `--redcap-api-batch-size` would be more self-explanatory without verging on verbosity? This (correctly) suggests a connection with the `REDCAP_API_…` environment variables too.

¹ CLI options are by overwhelming convention hyphen-separated, not underscore separated, so I assume that's what you meant.

Reply:
We configure the cron jobs with the options one time. It doesn't seem like a huge imposition to make longer option names.

Reply:
Oh it's certainly not a "huge imposition", but I think it's important to consider the user experience of the tools we write which are for primarily for humans (not cron).

Reply:
I think you're both talking about improving the UI for humans but differ about what that means. To @jccraft-uw, `--limit` is too vague and `--receiving-limit` more clearly indicates its purpose. For @kairstenfay, `--limit` is clear while `--receiving-limit` seems to add an unnecessary and wordy detail.

I'm ambivalent here. `--limit` is what made sense to me originally, but I'm ok to make it `--receiving-limit` instead.

If there's more people want to say here, then I'm open to continuing the conversation for a little more time, but if things are at an impasse, then I don't want to get too bogged down in this decision. If that's the case, then either I can pick one option or support both.

Reply:
I can look at the code if I forget what each option does. I'm OK with short names since that is the team norm.

Reply:
I'm mostly ambivalent here, too, and I certainly don't want to create an impasse. I will happily accept whatever the outcome here, whether we change one, both, or neither option names discussed in this thread. 

I just wanted to express my support for @tsibley's naming an option "limit" that's as close to the SQL as possible; but also, I agree with @jccraft-uw  that `--fetch-batch-size` has room for improvement (naming is consistently one of the hardest parts of programming in my experience).

@jccraft-uw If one of us needs to dig into the code to figure out what an option means, then we've created unusable tools. We should try to avoid that at all costs! Would the help message be useful to you? If you run any of our id3c commands with `--help`, you should get a nice description of what the option does. There's a lot of room for verbosity there.

Reply:
It would be nice for the options to be clear enough that one doesn't need to read the help to know what they do.
Again, though, I withdraw my request.

Reply:
Ok.

Here's what I've done, as a compromise that I feel good about:

- `--limit` is now `--det-limit`
- `--fetch-batch-size` is now `--redcap-api-batch-size`

I'm going to merge this and if there are further concerns, then we can address those separately.